src/libostree/ostree-impl-system-generator.c

```diff
@@ -261,17 +262,19 @@ _ostree_impl_system_generator (const char *normal_dir, const char *early_dir, co
   if (!cmdline)
     return glnx_throw (error, "Failed to read /proc/cmdline");
 
-  g_autofree char *ostree_cmdline = otcore_find_proc_cmdline_key (cmdline, "ostree");
-
+  g_autoptr (GError) otcore_get_ostree_target_error = NULL;
```

Comment:
We don't actually care so much if otcore_get_ostree_target fails but if there is no karg found for ostree selection we should return true for the CoreOS live environments.

Reply:
If the androidboot karg doesn't exist all behavior is identical to what this code was like previous to this commit...

Reply:
Or are we saying:

```
  if (!otcore_get_ostree_target (cmdline, &is_aboot, &ostree_target,
                                 NULL)
```

should work? Didn't do this just in case of NULL ptr dereference or something, but if you are confident this works ok I trust you :smile: